### ISSUE TITLE
scikit-image wheels includes the doc module

### ISSUE THREAD

**Issue by rth** [scikit-image/scikit-image#6505]
Currently, the scikit-image wheels include  files from the doc/ folder,
```
$ unzip scikit_image-0.19.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
...
   creating: doc/ext/
   creating: doc/ext/tests/
  inflating: doc/ext/__init__.py     
  inflating: doc/ext/plot2rst.py     
  inflating: doc/ext/doi_role.py     
  inflating: doc/ext/notebook_doc.py  
  inflating: doc/ext/tests/test_notebook_doc.py  
  inflating: doc/ext/tests/__init__.py
```
and it also records `doc` as a module installed by scikit-image in [`top_level.txt`](https://setuptools.pypa.io/en/latest/deprecated/python_eggs.html#top-level-txt-conflict-management-metadata) which is not ideal,
```
$ cat scikit_image-0.19.3.dist-info/top_level.txt
doc
skimage
```

Identified by @ryanking13 while packaging scikit-image in Pyodide.

**Comment by mkcor** [comment]
This seems intentional:
https://github.com/scikit-image/scikit-image/blob/6342ec594fb48126013305155d5643a3d9764557/MANIFEST.in#L15-L22
but I'm not very knowledgeable about this.

/cc @grlee77 @hmaarrfk @lagru

**Comment by rth** [comment]
MANIFEST.in is responsible for the files in the .tar.gz (and including docs/ there could make sense) but I'm not sure it affects the wheels (as for instance the .rst files specified in the manifest are not included in the wheel).

**Comment by lagru** [comment]
Hmm, according to [Controlling files in the distribution](https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html) we are using the correct option `include_package_data=False` so that files from MANIFEST.in are not included in the wheels. Because only *.py files are included I'm guessing that maybe the `setuptools.find_packages(exclude=['doc', 'doc.*', 'benchmarks'])` call is not working as intended?

**Comment by lagru** [comment]
@rth thanks for the report!

**Comment by lagru** [comment]
Looking at PyPI, I think this started with 0.19.0rc0. Before that the doc/ext is not included in the wheels (didn't check everything obviously :sweat_smile: ).

**Comment by lagru** [comment]
Aha, I think this was already fixed in #6428. We just haven't released new wheels with that fix yet. I verified locally with [Greg's helpful comment](https://github.com/scikit-image/scikit-image/pull/6428#pullrequestreview-1029430130) that this fix still works to exclude the docs dir locally.

Offtopic, at some point we should probably discuss using a src-layout for our repository.

**Comment by rth** [comment]
Great, thanks for investigating! Closing as resolved then.
